tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,329 @@
+import argparse
+import logging
+from os.path import dirname
+import matplotlib.pyplot as plt
+
+import hail as hl
+
+from gnomad.utils.reference_genome import get_reference_genome
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("sex_check")
+logger.setLevel(logging.INFO)
+
+
+def get_chr_cov(
+    mt: hl.MatrixTable,
+    build: str,
+    chr_name: str,
+    af_field: str = "AF",
+    call_rate_threshold: float = 0.25,
+    af_threshold: float = 0.01,
+) -> hl.Table:
+    """
+    Calculate mean chromosome coverage.
+
+    :param mt: MatrixTable containing samples with chrY variants
+    :param build: Reference used, either GRCh37 or GRCh38
+    :param chr_name: Chosen chromosome. Must be either autosome (number only) or sex chromosome (X, Y)
+    :param af_field: Name of field containing allele frequency information. Default is "AF"
+    :param call_rate_threshold: Minimum call rate threshold. Default is 0.25
+    :param af_threshold: Minimum allele frequency threshold. Default is 0.01
+    :return: Table with coverage annotations
+    """
+
+    logger.warning(
+        "This function expects the chrY to be at index 23 and chrX to be at index 22."
+    )
+
+    if chr_name == "Y":
+        filter_nonpar_expr = sex_mt.locus.in_y_nonpar()
+        chr_place = 23
+    elif chr_name == "X":
+        filter_nonpar_expr = sex_mt.locus.in_x_nonpar()
+        chr_place = 22
+    else:
+        try:
+            # Chromosome index in '.contigs' list should be one less than the chromosome number
+            chr_place = int(chr_name) - 1
+        except ValueError:
+            return logger.error("chr_name cannot be converted to an integer")
```

Comment:
```suggestion
            logger.error("chr_name cannot be converted to an integer")
            return -99
```
updating this so that the expression used to normalize chromosome coverage will still work